### PR TITLE
Validate `transport.profiles.*` settings

### PR DESCRIPTION
**Pull request by s1monw**
Transport profiles unfortunately have never been validated. Yet, it's very
easy to make a mistake when configuring profiles which will most likely stay
undetected since we don't validate the settings but allow almost everything
based on the wildcard in `transport.profiles.*`. This change removes the
settings subset based parsing of profiles but rather uses concrete affix settings
for the profiles which makes it easier to fall back to higher level settings since
the fallback settings are present when the profile setting is parsed. Previously, it was
unclear in the code which setting is used ie. if the profiles settings (with removed
prefixes) or the global node setting. There is no distinction anymore since we don't pull
prefix based settings.

### CODE REVIEW

**Review by rjernst — APPROVED** [review]
LGTM

File: modules/transport-netty4/src/test/java/org/elasticsearch/transport/netty4/NettyTransportMultiPortTests.java [hunk 1]
```diff
@@ -92,9 +92,9 @@ public void testThatProfileWithoutPortSettingsFails() throws Exception {
             .build();
 
         ThreadPool threadPool = new TestThreadPool("tst");
-        try (TcpTransport<?> transport = startTransport(settings, threadPool)) {
-            assertEquals(0, transport.profileBoundAddresses().size());
-            assertEquals(1, transport.boundAddress().boundAddresses().length);
+        try {
+            IllegalStateException ex = expectThrows(IllegalStateException.class, () -> startTransport(settings, threadPool));
+            assertEquals("profile [client1] has not port configured", ex.getMessage());
```
Comment (jaymode) [inline modules/transport-netty4/src/test/java/org/elasticsearch/transport/netty4/NettyTransportMultiPortTests.java]: s/not/no

File: core/src/main/java/org/elasticsearch/transport/TcpTransport.java [hunk 2]
```diff
@@ -1729,4 +1705,61 @@ public final TransportStats getStats() {
...
+            reuseAddress = TCP_REUSE_ADDRESS_PROFILE.getConcreteSettingForNamespace(profileName).get(settings);
+            sendBufferSize = TCP_SEND_BUFFER_SIZE_PROFILE.getConcreteSettingForNamespace(profileName).get(settings);
+            receiveBufferSize = TCP_RECEIVE_BUFFER_SIZE_PROFILE.getConcreteSettingForNamespace(profileName).get(settings);
+            List<String> profileBindHosts = BIND_HOST_PROFILE.getConcreteSettingForNamespace(profileName).get(settings);
+            bindHosts = (profileBindHosts.isEmpty() ? NetworkService.GLOBAL_NETWORK_BINDHOST_SETTING.get(settings)
+                : profileBindHosts);
+            publishHosts = PUBLISH_HOST_PROFILE.getConcreteSettingForNamespace(profileName).get(settings);
+            Setting<String> concretePort = PORT_PROFILE.getConcreteSettingForNamespace(profileName);
+            if (concretePort.exists(settings) == false && isDefaultProfile == false) {
+                throw new IllegalStateException("profile [" + profileName + "] has not port configured");
```
Comment (jaymode) [inline core/src/main/java/org/elasticsearch/transport/TcpTransport.java]: s/not/no

**Review by jaymode — APPROVED** [review]
LGTM

**Pull request merged**